crypto3/libs/blueprint/include/nil/blueprint/bbf/circuit_builder.hpp

```diff
@@ -0,0 +1,668 @@
+                                        if (!found) {
+                                            std::cout << "Input values:";
+                                            for (std::size_t k = 0; k < input_values.size(); k++) {
+                                                std::cout << input_values[k] << " ";
+                                            }
+                                            std::cout << std::endl;
+                                            std::cout << "Constraint " << j << " from lookup gate " << i << " from table "
+                                                << table_name << " on row " << selector_row << " is not satisfied."
+                                                << std::endl;
+                                            std::cout << "Offending Lookup Gate: " << std::endl;
+                                            for (const auto &constraint : lookup_gates[i].constraints) {
+                                                std::cout << "Table id: " << constraint.table_id << std::endl;
+                                                for (auto &lookup_input : constraint.lookup_input) {
+                                                    std::cout << lookup_input << std::endl;
+                                                }
+                                            }
+                                            return false;
+                                        }
+                                    } catch (std::out_of_range &e) {
+                                        std::cout << "Lookup table " << table_name << " not found." << std::endl;
+                                        std::cout << "Table_id = " << lookup_gates[i].constraints[j].table_id << " table_name "
+                                                  << table_name << std::endl;
+                                        return false;
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    for (const auto& i : used_copy_constraints) {
+                        if (var_value(assignments, copy_constraints[i].first) !=
+                            var_value(assignments, copy_constraints[i].second)) {
+                            std::cout << "Copy constraint number " << i << " is not satisfied."
+                                      << " First variable: " << copy_constraints[i].first
+                                      << " second variable: " << copy_constraints[i].second << std::endl;
+                            std::cout << var_value(assignments, copy_constraints[i].first) << " != "
+                                      << var_value(assignments, copy_constraints[i].second) << std::endl;
+                            return false;
+                        }
+                    }
+                    return true;
+                }
+
+                bool check_proof(
```

Comment:
I believe this function belongs to the test, not this class. Blueprint probably must not be aware of the prover.

Reply:
Moved function to test.